src/alignment.jl

```diff
@@ -100,41 +104,74 @@ function Base.show(io::IO, aln::Alignment)
       print(io, "  CIGAR string: ", cigar(aln))
 end
 
-"""
-    seq2ref(aln::Alignment, i::Integer)::Tuple{Int,Operation}
-
-Map a position `i` from sequence to reference.
-"""
-function seq2ref(aln::Alignment, i::Integer)::Tuple{Int,Operation}
-    idx = findanchor(aln, i, Val{true})
+# generic function for mapping between sequence, reference and alignment positions
+# getsrc specifies anchor source position getter
+# getdest specifies anchor destination position getter
+function pos2pos(aln::Alignment, i::Integer,
+                 srcpos::Function, destpos::Function)::Tuple{Int,Operation}
+    idx = findanchor(aln, i, srcpos)
     if idx == 0
-        throw(ArgumentError("invalid sequence position: $i"))
+        if srcpos === seqpos
```

Comment:
It's not a variable, it's a function defined in `anchors.jl` that retrieves a specific position.
That's why I'm using `===`. It relates to your other comments below.

The function references serve here both as functions that get the positions and as parameters that specify how counting should be done.
The motivation was to reduce code duplication that was there before (`seq2ref()` and `ref2seq()`),
because when introducing aligned position and mapping between all 3 types of positions, the code duplication would have been further increased.